src/handlers/tracker.ts

```diff
@@ -71,6 +97,34 @@ export class TrackerHandler implements IDisposable {
     });
   }
 
+  protected onCurrentSourceOpened(_: Sources.Model, source: Sources.ISource) {
+    if (!source) {
+      return;
+    }
+    const debugSessionPath = this.debuggerService.session.client.path;
+    const { code, mimeType, path } = source;
+    const results = this.find(debugSessionPath, path);
+    if (results.next()) {
+      return;
+    }
+    const editorWrapper = this.readOnlyEditorFactory.createNewEditor({
+      code,
+      mimeType,
+      path
+    });
+    const editorHandler = new EditorHandler({
+      debuggerService: this.debuggerService,
+      editor: editorWrapper.editor
+    });
+    const widget = new Panel();
+    widget.addWidget(editorWrapper);
+    widget.id = UUID.uuid4();
```

Comment:
Maybe `path` could be used for `widget.id`.

Reply:
This can create a potential bug if the first character of the generated string is a number, which is an invalid UI in HTML. In JupyterLab, we created a helper function to guarantee this never happens: `DOMUtils.createDomID()` in the `@jupyterlab/apputils` library.

Reply:
This is indeed better, thanks.